cli/azd/pkg/tools/azcli.go

```diff
@@ -617,6 +626,23 @@ func (cli *azCli) ListResourceGroupResources(ctx context.Context, subscriptionId
 	return resources, nil
 }
 
+func (cli *azCli) GetResource(ctx context.Context, subscriptionId string, resourceId string) (AzCliResourceExtended, error) {
+	res, err := cli.runAzCommand(ctx, "resource", "show", "--ids", resourceId, "--output", "json")
+	if isNotLoggedInMessage(res.Stderr) {
+		return AzCliResourceExtended{}, ErrAzCliNotLoggedIn
+	} else if err != nil {
+		return AzCliResourceExtended{}, fmt.Errorf("failed running az resource show --ids: %s: %w", res.String(), err)
+	}
+
+	var resource AzCliResourceExtended
+
+	if err := json.Unmarshal([]byte(res.Stdout), &resource); err != nil {
+		return AzCliResourceExtended{}, fmt.Errorf("could not unmarshal output %s as a AzCliResourceExtended: %w", res.Stdout, err)
+	}
+
+	return resource, nil
+}
```

Comment:
I recommend you move this func into `ResourceManager` since the `AzCli` tool should really only be lightweight shim calls into the az cli without any additional processing.

Reply:
Hmm, the way this function is written correctly is indeed a lightweight shim call. It calls `az resource show` and unmarshals it into a structured object.